app/services/host_inventory_api.rb

```diff
@@ -11,7 +11,7 @@ def initialize(id, hostname, account, url, b64_identity)
     @hostname = hostname
     @url = "#{URI.parse(url)}#{ENV['PATH_PREFIX']}/inventory/v1/hosts"
     @account = account
-    @b64_identity = b64_identity
+    @b64_identity = b64_identity || @account.b64_identity
```

Comment:
You could probably make then `b64_identity` an optional argument from now on

Reply:
It's implicitly optional (you may pass nil) like this. I was trying to avoid changing the constructor signature and every usage in the code base, but I can do that if you'd like.